### DISCUSSION TITLE
Name Conflict Error While Importing Into ThingWorx

### DISCUSSION THREAD

**Discussion by vinesh-parthasarathy** [thingworx-field-work/ThingworxVSCodeProject#47] [General]
Requirement is to update the default value of a template configuration. **@ConfigurationTables** and **@config** annotations were used on a thing template and required changes were made . But it throws a Name Conflict error while importing the extension with the changes into thingworx . While checking about the error, we found that template inherits the configuration from a parent template . To set the config value ,value has to be overridden in the child template level. so is there any way to override child template and set the default values in configuration and I also attach the screenshot of error.

![image](https://user-images.githubusercontent.com/94957050/159907226-1437bdfe-bdc4-4a03-87db-2443e38531b0.png)

**Comment by BogdanMihaiciuc** [comment]
Hey,

Once an extension entity with a configuration table has been imported, subsequent updates to the extension will not change configuration table values. This is standard thingworx behavior applicable to all extensions.

To change these values with an update, you will need a service to perform these changes.

Regarding the two decorators, use **ConfigurationTables** to declare configuration tables. You can’t override these in subclasses. Use **config** to set configuration table values. You can also use this to set values for inherited tables.

Hope this clears it up.

**Comment by vinesh-parthasarathy** [comment]
@BogdanMihaiciuc  Can you guide as how to add a subscription using typescript to thingworx platform.

**Comment by BogdanMihaiciuc** [comment]
See https://github.com/ptc-iot-sharing/ThingworxVSCodeProject/blob/418f9f9ef0cf4c7d0fbcf56a40a299cce50c1a19/src/MyThing.ts#L143 and https://github.com/ptc-iot-sharing/ThingworxVSCodeProject/blob/418f9f9ef0cf4c7d0fbcf56a40a299cce50c1a19/src/MyThing.ts#L157
